src/sensors/src/ThreeAxisAngularAccelerometerSensor.cpp

```diff
@@ -0,0 +1,181 @@
+ * Public License for more details
+ */
+
+
+# include "iDynTree/Core/LinearMotionVector3.h"
+
+#include "iDynTree/Sensors/ThreeAxisAngularAccelerometerSensor.h"
+
+#include "iDynTree/Core/Transform.h"
+
+#include "iDynTree/Core/SpatialAcc.h"
+#include "iDynTree/Core/Twist.h"
+
+namespace iDynTree {
+
+struct ThreeAxisAngularAccelerometerSensor::ThreeAxisAngularAccelerometerPrivateAttributes
+{
+    // Name/id of the sensor
+    std::string name;
+    // Transform from the link to the sensor
+    Transform link_H_sensor;
+    // Index of the parent junction
+    int parent_link_index;
+    // Name of the parent junction
+     std::string parent_link_name;
+    // Name of the link to which the Accelerometer is connected
+};
+
+
+ThreeAxisAngularAccelerometerSensor::ThreeAxisAngularAccelerometerSensor()
+{
+    this->pimpl = new ThreeAxisAngularAccelerometerPrivateAttributes;
+
+    this->pimpl->name = "";
+    this->pimpl->link_H_sensor = Transform::Identity();
+    this->pimpl->parent_link_index = -1;
+    this->pimpl->parent_link_name = "";
+
+}
+
+ThreeAxisAngularAccelerometerSensor::ThreeAxisAngularAccelerometerSensor(const ThreeAxisAngularAccelerometerSensor& other):
+    pimpl(new ThreeAxisAngularAccelerometerPrivateAttributes(*(other.pimpl)))
+{
+
+}
+
+ThreeAxisAngularAccelerometerSensor& ThreeAxisAngularAccelerometerSensor::operator=(const ThreeAxisAngularAccelerometerSensor& other)
+{
+    if(this != &other)
+    {
+        *pimpl = *(other.pimpl);
+    }
+    return *this;
+}
+
+
+ThreeAxisAngularAccelerometerSensor::~ThreeAxisAngularAccelerometerSensor()
+{
+    delete this->pimpl;
+}
+
+bool ThreeAxisAngularAccelerometerSensor::setName(const std::string& _name)
+{
+    this->pimpl->name = _name;
+    return true;
+}
+
+
+bool ThreeAxisAngularAccelerometerSensor::setLinkSensorTransform(const iDynTree::Transform& link_H_sensor)
+{
+      this->pimpl->link_H_sensor = link_H_sensor;
+      return true;
+}
+
+
+bool ThreeAxisAngularAccelerometerSensor::setParentLink(const std::string& parent)
+{
+    this->pimpl->parent_link_name = parent;
+    return true;
+}
+
+bool ThreeAxisAngularAccelerometerSensor::setParentLinkIndex(const LinkIndex &parent_index)
+{
+    this->pimpl->parent_link_index = parent_index;
+    return true;
+
+}
+
+std::string ThreeAxisAngularAccelerometerSensor::getParentLink() const
+{
+    return(this->pimpl->parent_link_name);
+}
+
+int ThreeAxisAngularAccelerometerSensor::getParentLinkIndex() const
+{
+    return(this->pimpl->parent_link_index);
+}
+
+bool ThreeAxisAngularAccelerometerSensor::isValid() const
+{
+    if( this->getName() == "" )
+    {
+        return false;
+    }
+
+    if( this->pimpl->parent_link_index < 0 )
+    {
+        // Return false because the links is not appropriately setted
+        return false;
+    }
+
+    return true;
+}
+
+Sensor* ThreeAxisAngularAccelerometerSensor::clone() const
+{
+    return (Sensor *)new ThreeAxisAngularAccelerometerSensor(*this);
```

Comment:
you want the classic warning on old C style cast to popup? 😂 
BTW: [this](https://stackoverflow.com/questions/259853/whats-the-best-signature-for-clone-in-c) suggests that the return type can be `ThreeAxisAngularAccelerometerSensor`

Reply:
See https://github.com/robotology/idyntree/pull/356#discussion_r135543256 .